### PR TITLE
feat(CLI): add support for shared projects

### LINKED ISSUE #230: Stryker lacks support for shared projects
My first attempts with Stryker on NFluent resulted in no mutants being generated.
After ruling out configuration error and looking at the codebase, I understood this was a limitation due to us relying on shared projects to list source files (see [this project file](https://github.com/tpierrain/NFluent/blob/master/src/NFluent.Standard.20/NFluent.Standard.20.csproj)).
I have worked on a first implementation for this feature, and I am ironing out tests before sendin a PR

### PR DESCRIPTION
**Pull request by dupdob**
Internal impact: creation of a placeholder Root folder to host all projects. It is not reported. 
Closes #230.

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs [hunk 1]
```diff
@@ -98,7 +99,138 @@ public void InputFileResolver_InitializeShouldCrawlFilesRecursively()
...
+                    { Path.Combine(_filesystemRoot, "TestProject", "TestProject.csproj"), new MockFileData(testProjectFile)},
+                    { Path.Combine(_filesystemRoot, "TestProject", "bin", "Debug", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                    { Path.Combine(_filesystemRoot, "TestProject", "obj", "Release", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                });
+
+            var target = new InputFileResolver(fileSystem);
+
+            var result = target.ResolveInput(Path.Combine(_filesystemRoot, "TestProject"), "");
+
+
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]: An enter too much

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs [hunk 2]
```diff
@@ -98,7 +99,138 @@ public void InputFileResolver_InitializeShouldCrawlFilesRecursively()
             var result = target.ResolveInput(Path.Combine(_filesystemRoot, "TestProject"), "");
 
             result.TestProjectPath.ShouldBe(Path.Combine(_filesystemRoot, "TestProject"));
-            result.ProjectContents.Children.Count.ShouldBe(2);
+            result.ProjectContents.GetAllFiles().Count().ShouldBe(2);
+        }
+
+        [Fact]
+        public void InputFileResolver_InitializeShouldSupportSharedProjects()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]:
```suggestion
        public void InputFileResolver_InitializeShouldResolveImportedProjects()
```

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs [hunk 3]
```diff
@@ -98,7 +99,138 @@ public void InputFileResolver_InitializeShouldCrawlFilesRecursively()
...
+                    { Path.Combine(_filesystemRoot, "TestProject", "TestProject.csproj"), new MockFileData(testProjectFile)},
+                    { Path.Combine(_filesystemRoot, "TestProject", "bin", "Debug", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                    { Path.Combine(_filesystemRoot, "TestProject", "obj", "Release", "netcoreapp2.0"), new MockFileData("Bytecode") },
+                });
+
+            var target = new InputFileResolver(fileSystem);
+
+            Assert.Throws<FileNotFoundException>(() =>
+                target.ResolveInput(Path.Combine(_filesystemRoot, "TestProject"), ""));
+
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]: I'm missing a test case for multiple shared projects included.

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs (hunk 1 above)
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]: Done

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs (hunk 2 above)
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]: Done

File: src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs (hunk 3 above)
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InputFileResolverTests.cs]: Done

**Comment by dupdob** [comment]
I don't understand why the build is failing on Azure. It works OK here. Error messages are not consistent with project settings.

**Comment by dupdob** [comment]
The build pipeline checks at the package publication chain, I let you revert to me if I have to do anything to fix this.

**Comment by richardwerkman** [comment]
@dupdob Sorry about this, we are currently working on improved integration tests. They seem to fail on all forked branches. We will try to fix this as soon as possible.

**Comment by dupdob** [comment]
Many thanks, looking forward to test NFluent with the next official release and not my fork

**Pull request merged**